apps/system/js/notifications_system_message.js

```diff
@@ -0,0 +1,157 @@
+        this.closeOldNotification(message);
+        return;
+      }
+
+      var target = message.data.systemMessageTarget;
+      if (!target) {
+        console.error('System message target seems undefined, force-closing');
+        this.closeOldNotification(message);
+        return;
+      }
+
+      if (!this.hasRegisteredTarget(target)) {
+        console.error('No module has subscribed to: ' + target);
+        return;
+      }
+
+      this.callRegisteredTarget(target, message);
+    },
+
+    registerTarget: function(target, obj) {
+      if (this.hasRegisteredTarget(target)) {
+        this.debug('Already have a handler for target: ' + target);
+        return;
+      }
+
+      this.debug('Registering a handler for target: ' + target);
+      this._handlers[target] = {
+        name: target,
+        context: obj
+      };
+    },
+
+    unregisterTarget: function(target, obj) {
+      if (!this.hasRegisteredTarget(target)) {
+        console.error('No handler to unregister for target: ' + target);
+        return;
+      }
+
+      var c = this.getTargetContext(target);
+      if (c !== obj) {
+        console.error('Mismatching object to unregister for target: ' + target);
+        return;
+      }
+
+      this.debug('Unregistering the handler for target: ' + target);
+      this._handlers[target] = null;
+      delete this._handlers[target];
+    },
+
+    callRegisteredTarget: function(target, payload) {
+      var ctx = this.getTargetContext(target);
+      if (!ctx) {
+        console.error('No registered target for ' + target);
+        return;
+      }
+
+      ctx.handleSystemMessageNotification.call(ctx, payload);
+    },
+
+    closeOldNotification: function(message) {
+      this.debug('Will try to close old notification: ' +
+                 JSON.stringify(message));
+
+      // Fields from system message notification may be undefined,
+      // while the same field in the Notification object will be empty
+      message.title = message.title ? message.title : '';
+      message.tag   = message.tag   ? message.tag   : '';
+      message.body  = message.body  ? message.body  : '';
+
+      Notification.get().then(notifs => {
+        notifs.forEach(notif => {
+          if (notif.data && notif.data.systemMessageTarget) {
+            return;
+          }
+
+          if ((notif.body  === message.body) &&
+              (notif.tag   === message.tag) &&
+              (notif.title === message.title)) {
+            console.warn('Found a match for "' + message.title + '": "' +
+                         message.body + '". Will force-close this.');
+            notif.close();
+          }
+        });
+      });
+    },
+
+    getTargetContext: function(target) {
+      return this.getRegisteredTarget(target).context;
+    },
+
+    getRegisteredTarget: function(target) {
+      return this._handlers[target];
+    },
+
+    hasRegisteredTarget: function(target) {
+      return (target in this._handlers);
+    },
+
```

Comment:
I am trying to comment this service in order to help other to use.
/**
  The service is used by other modules who seeks for specific notifications. There is only one module for one target at one time. The module should provide a method named for handleSystemMessageNotification(); it will be invoked once we get the notification via system message which data.systemMessageTarget is the same as the target specified.
 */